ibridges/session.py

```diff
@@ -240,36 +212,30 @@ def server_version(self) -> tuple:
 
         """
         try:
-            return self._irods_session.server_version
+            return self.irods_session.server_version
         except Exception as e:
-            if repr(e) in LoginFail:
-                raise ValueError(LoginFail[repr(e)]) from e
-            raise e
-
-    @property
-    def username(self) -> str:
-        """Retrieve username.
-
-        Returns
-        -------
-        str
-            Username.
-
-        """
-        if self._irods_session:
-            return self._irods_session.username
-        return ''
-
-    @property
-    def zone(self) -> str:
-        """Retrieve the zone name.
-
-        Returns
-        -------
-        str
-            Zone.
-
-        """
-        if self._irods_session:
-            return self._irods_session.zone
-        return ''
+            raise _translate_irods_error(e) from e
+
+
+class LoginError(ValueError):
+    """Error indicating a failure to log into the iRods server."""
+
+
+
+def _translate_irods_error(exc) -> Exception:  # pylint: disable=too-many-return-statements
+    if isinstance(exc, NetworkException) and exc.msg.startswith(
+            'Client-Server negotiation failure'):
+        return LoginError("Host, port, irods_client_server_policy or "
+                          "irods_client_server_negotiation not set correctly in "
+                          "irods_environment.json")
+    if isinstance(exc, CAT_INVALID_USER):
```

Comment:
same for the PAM_AUTH_PASSWORD_FAILED etc. The translation does not work, since the Exception is not of instance `PAM_AUTH_PASSWORD_FAILED`. That was why I mapped the whole repr(Exception) to something.

Reply:
Can you try again? I do think it should work now. I think it was just that using the `server_version` property was raising the error and that was outside of the scope of the catch.

Not sure what it is with the integration test though..